protocol/x/clob/keeper/authenticators.go

```diff
@@ -0,0 +1,73 @@
+package keeper
+
+import (
+	errorsmod "cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	authsigning "github.com/cosmos/cosmos-sdk/x/auth/signing"
+	accountpluslib "github.com/dydxprotocol/v4-chain/protocol/x/accountplus/lib"
+	aptypes "github.com/dydxprotocol/v4-chain/protocol/x/accountplus/types"
+	"github.com/dydxprotocol/v4-chain/protocol/x/clob/types"
+)
+
+// MaybeValidateAuthenticators checks if the transaction has authenticators specified and if so,
+// validates them. It returns an error if the authenticators are not valid or removed from state.
+func (k Keeper) MaybeValidateAuthenticators(ctx sdk.Context, txBytes []byte) error {
+	// Decode the tx from the tx bytes.
+	tx, err := k.txDecoder(txBytes)
```

Comment:
We need to decode maker orders for all fills (regardless of smart account) - do we think this is okay because there are not many fews?

Reply:
I was planning to look at frame graphs after this is merged and deployed and optimize only if needed. I feel like this might be okay since like you said, only a very small percentage of the orders get matched.

note that all crossing maker orders get decoded, not just those that generate a fill